### PR TITLE
daos-7022 documentation: Update docs and github.io site

### PR DESCRIPTION
**Pull request by cdurf1**
Skip-build: true
Skip-test: true

Signed-off-by: craig <craig.durfey@intel.com>

### CODE REVIEW

**Review by tanabarr — CHANGES_REQUESTED** [review]
I'm a bit confused as to why there are references to local user paths in C:/ home drives in this PR, wouldn't expect any user specific references in a DAOS PR

**Comment by johannlombardi** [comment]
Yeah, the doxyfile is something that only Craig is able to use .... that's indeed a bummer.

**Review by tanabarr — APPROVED** [review]

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        - 'Support Matrix': 'release/support_matrix.md'
+    - Developer Zone:
+        - 'Dev Environment': 'dev/development.md'
+        - 'Contributing': 'dev/contributing.md'
+        - 'DAOS Internals': 'https://github.com/daos-stack/daos/blob/master/src/README.md'
+        - 'DAOS API Documentation': 'html/index.html'
+    - Community Wiki: 'https://wiki.hpdd.intel.com'
+    - Roadmap: 'https://wiki.hpdd.intel.com/display/DC/Roadmap/'
+    - Mailing list: 'https://daos.groups.io/g/daos'
+    - Issue tracker: 'https://jira.hpdd.intel.com/projects/DAOS'
```
Comment (johannlombardi) [inline mkdocs.yml:74]: Please use doc/mkdocks.yml instead.

File: Doxyfile [hunk 2]
```diff
@@ -914,15 +925,8 @@ RECURSIVE              = YES
...
-                         src/include/daos \
-                         src/include/daos_srv \
-                         src/include/gurt \
-                         src/include/daos.h \
-                         src/include/daos_mgmt.h \
-                         src/include/daos_pool.h \
-                         src/include/daos_prop.h \
-                         src/include/dfuse_ioctl.h
+EXCLUDE                = daos_cdurf/src/include/daos.h \
+                         daos_cdurf/src/include/daos_mgmt.h
```
Comment (johannlombardi) [inline Doxyfile:929]: should not include daos_cdurf since INPUT above does not.

File: debug.log [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[0324/173033.566:ERROR:registration_protocol_win.cc(130)] TransactNamedPipe: The pipe has been ended. (0x6D)
+[0324/173242.135:ERROR:registration_protocol_win.cc(130)] TransactNamedPipe: The pipe has been ended. (0x6D)
```
Comment (johannlombardi) [inline debug.log:2]: Debug logs should not be checked in.

**Review by johannlombardi — CHANGES_REQUESTED** [review]
Still not ready.

**Pull request closed without merging**